server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -92,10 +102,15 @@ public DeprecationLogger warn(final DeprecationCategory category, final String k
     }
 
     private DeprecationLogger logDeprecation(Level level, DeprecationCategory category, String key, String msg, Object[] params) {
-        String opaqueId = HeaderWarning.getXOpaqueId();
-        String productOrigin = HeaderWarning.getProductOrigin();
-        ESLogMessage deprecationMessage = new DeprecatedMessage(category, key, opaqueId, productOrigin, msg, params);
-        logger.log(level, deprecationMessage);
+        List<String> skipTheseDeprecations = initialEnvironment == null
+            ? Collections.emptyList()
+            : initialEnvironment.getAsList("deprecation.skip_deprecated_settings");
```

Comment:
Rather than re-create this list every time, and since we were planning on removing the dynamic property from `deprecation.skip_deprecated_settings` (since it's not really dynamic), can we just calculate the `List<String>` inside of `setInitialEnvironmentSettings` and then not have to calculate it for every log message?